mathesar_ui/src/pages/admin-users/UserListingPage.svelte

```diff
@@ -1,34 +1,147 @@
 <script lang="ts">
   import { getUsersStoreFromContext } from '@mathesar/stores/users';
+  import { ADMIN_USERS_PAGE_ADD_NEW_URL } from '@mathesar/routes/urls';
   import {
-    getEditUsersPageUrl,
-    ADMIN_USERS_PAGE_ADD_NEW_URL,
-  } from '@mathesar/routes/urls';
+    AnchorButton,
+    Button,
+    Icon,
+    iconSearch,
+    TextInputWithPrefix,
+  } from '@mathesar/component-library';
+  import { iconAddNew } from '@mathesar/icons';
+  import type { User } from '@mathesar/api/users';
+  import { labeledCount } from '@mathesar/utils/languageUtils';
+  import UserRow from './UserRow.svelte';
+
+  let filterQuery = '';
 
   const usersStore = getUsersStoreFromContext();
   $: requestStatus = $usersStore?.requestStatus;
   $: users = $usersStore?.users;
-  $: usersCount = $usersStore?.count;
+
+  function filterUsers(_users: User[], query: string) {
+    const isMatch = (user: User, q: string) =>
+      user.username.toLowerCase().includes(q) ||
+      user.full_name?.toLowerCase().includes(q) ||
+      user.email?.toLowerCase().includes(q);
+    return _users?.filter((user) => {
+      if (query) {
+        const sanitizedQuery = query.trim().toLowerCase();
+        return isMatch(user, sanitizedQuery);
+      }
+      return true;
+    });
+  }
+
+  function handleClearFilterQuery() {
+    filterQuery = '';
+  }
+
+  $: filteredUsers = filterUsers($users ?? [], filterQuery);
 </script>
 
-{#if $requestStatus?.state === 'processing'}
-  Loading
-{:else if $requestStatus?.state === 'success'}
-  <div>{$usersCount} users</div>
-  <div>
-    <a class="btn btn-primary" href={ADMIN_USERS_PAGE_ADD_NEW_URL}>
-      Add user
-    </a>
-  </div>
-  <div>
-    {#each $users ?? [] as user}
-      <div>
-        <a href={getEditUsersPageUrl(user.id)}>
-          {user.full_name}
-        </a>
+<section class="users-list-container">
+  {#if $requestStatus?.state === 'processing'}
+    <p>Loading...</p>
+  {:else if $requestStatus?.state === 'success'}
+    <h2>
```

Comment:
I'd like for this to be an `h1` element, and I like for the component to avoid targeting this element with any CSS. We should set that CSS globally instead.

Reply:
Changed to `h1` in https://github.com/centerofci/mathesar/pull/2423/commits/752484b9913013e47f56660c1bab519488169c2e. I think the global CSS change should be in a different PR, so that part is left the same. I noticed that you made some changes for the h1 global css in your PR for users edit/add page, we can take it up there.